pkg/test/environment/environment.go

```diff
@@ -49,6 +49,13 @@ type Interface interface {
 	// GetPilotOrFail returns a deployed Pilot instance in the environment, or fails the test if unsuccessful.
 	GetPilotOrFail(t testing.TB) DeployedPilot
 
+	// GetAPIServer returns a handle to the ambient API Server in the environment.
+	GetAPIServer() (DeployedAPIServer, error)
```

Comment:
Since APIServer is a k8s construct, I don't think it should appear at this level.  I suspect a better approach would be to give the components that need it access to the k8s environment directly. The API server is just part of a k8s environment ... it's always there, so it's not really a dependency in the same sense as what we've had previously.